### DISCUSSION TITLE
Using output variable within same step

### DISCUSSION THREAD

**Discussion by ivanaguilario** [argoproj/argo-workflows#8086] [Q&A] [answered]
Is it possible to reference other output parameters within the same step?

I need to reference an output parameter that loads some file contents so I can use those as the name for an artifact in the same step.

    apiVersion: argoproj.io/v1alpha1
    kind: Workflow
    metadata:
      generateName: work-
    spec:
      entrypoint: my-step
      templates:
      - name: my-step
        volumes:
        - name: artifacts
          emptyDir: { }
        outputs:
          parameters:
          - name: version
            valueFrom:
              path: /artifacts/version.txt
          artifacts:
          - name: artifacts
            path: /artifacts
            archive:
              none: {}
            s3:
              key: "" # How can I access the version output parameter here?
        container:
          image: my-container
          volumeMounts:
          - name: artifacts
            mountPath: /artifacts

**Comment by terrytangyuan** [comment, marked as answer]
An output variable can only be used in a next step. There’s data template feature that might help with your task: https://github.com/argoproj/argo-workflows/blob/master/examples/data-transformations.yaml

**Reply by ivanaguilario** [reply]
Thanks! It's unfortunate, but decent workaround at least for now.
